MacOS 13 linker parameter fail

|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.8.0 | Wazuh component | Agent | Sources | Mac OS 13 |

## Description
In the context of the tests carried out when creating/modifying a PR, this bad behavior arises in the `build-ventura` step, which at the time of compilation presents a problem with a linker parameter 
```shell
ld: unknown options: -noall_load
``` 

I share part of the output log
```shell
Makefile:2602: warning: overriding commands for target `win32/ui_resource.o'
Makefile:2542: warning: ignoring old commands for target `win32/ui_resource.o'
Makefile:2605: warning: overriding commands for target `win32/auth_resource.o'
Makefile:2545: warning: ignoring old commands for target `win32/auth_resource.o'
/Applications/Xcode_15.0.1.app/Contents/Developer/usr/bin/make build_sysinfo build_shared_modules build_syscollector
grep: /etc/os-release: No such file or directory
Makefile:2602: warning: overriding commands for target `win32/ui_resource.o'
Makefile:2542: warning: ignoring old commands for target `win32/ui_resource.o'
Makefile:2605: warning: overriding commands for target `win32/auth_resource.o'
Makefile:2545: warning: ignoring old commands for target `win32/auth_resource.o'
    CC libwazuhext.dylib
ld: unknown options: -noall_load 
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [libwazuhext.dylib] Error 1
make: *** [agent] Error 2
Error: Process completed with exit code 2.
```

--- a/src/Makefile
+++ b/src/Makefile
@@ -1870,7 +1870,7 @@ ifeq (${uname_S},Darwin)
 WAZUH_SHFLAGS=-install_name @rpath/libwazuhext.$(SHARED)
 
 $(WAZUHEXT_LIB): $(EXTERNAL_LIBS)
-	$(OSSEC_SHARED) $(OSSEC_CFLAGS) $(WAZUH_SHFLAGS) -o $@ -Wl,-all_load $^ -Wl,-noall_load $(OSSEC_LIBS)
+	$(OSSEC_SHARED) $(OSSEC_CFLAGS) $(WAZUH_SHFLAGS) -o $@ -Wl,-all_load $^ $(OSSEC_LIBS)
 else
 ifeq (${TARGET}, winagent)
 $(WAZUHEXT_LIB): $(WAZUHEXT_DLL) $(WAZUHEXT_LIB_DEF)
@@ -2102,7 +2102,7 @@ ifeq (${uname_S},Darwin)
 WAZUH_SHARED_SHFLAGS=-install_name @rpath/libwazuhshared.$(SHARED)
 
 $(WAZUH_LIB): $(WAZUHEXT_LIB) $(AR_PROGRAMS_DEPS)
-	$(OSSEC_SHARED) $(OSSEC_CFLAGS) $(WAZUH_SHARED_SHFLAGS) -o $@ -Wl,-all_load $^ -Wl,-noall_load $(OSSEC_LIBS)
+	$(OSSEC_SHARED) $(OSSEC_CFLAGS) $(WAZUH_SHARED_SHFLAGS) -o $@ -Wl,-all_load $^ $(OSSEC_LIBS)
 else
 ifeq (${TARGET}, winagent)
 $(WAZUH_DLL) $(WAZUH_DEF) : $(WAZUHEXT_DLL) $(AR_PROGRAMS_DEPS) win32/version-dll.o

